### PR TITLE
Feature/area rotation

### PR DESCRIPTION
**Pull request by cpittet**
This PR introduces a new feature for the area in Google Map : you can now rotate the rectangles so that it can better match physical areas.

The tests cover the small functions used to make the computations for the rotations.

Note : this PR must be merged after Mathieu's one.

Thanks for your review and comments :)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -34,6 +34,9 @@ android {
     kotlinOptions {
         jvmTarget = '1.8'
     }
+    //testOptions {
+    //    unitTests.returnDefaultValues = true
+    //}
```
Comment (Mathematou) [inline app/build.gradle:39]: I deleted this since I didn't use it

File: app/src/main/java/com/github/sdpteam15/polyevents/fragments/MapsFragment.kt [hunk 2]
```diff
@@ -79,14 +83,14 @@ class MapsFragment : Fragment(), OnMapReadyCallback, OnPolylineClickListener,
...
-        GoogleMapHelper.map = googleMap
-
-        GoogleMapHelper.map!!.setOnPolylineClickListener(this)
-        GoogleMapHelper.map!!.setOnPolygonClickListener(this)
-        GoogleMapHelper.map!!.setOnMarkerClickListener(this)
-        GoogleMapHelper.map!!.setOnMarkerDragListener(this)
-        GoogleMapHelper.map!!.setOnInfoWindowClickListener(this)
-        GoogleMapHelper.map!!.setOnMyLocationButtonClickListener(this)
+        GoogleMapHelper.map = GoogleMapAdapter(googleMap)
+        //GoogleMapHelper.map = googleMap
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/fragments/MapsFragment.kt:87]: I think we do not need to keep this line :)

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 3]
```diff
@@ -261,46 +282,155 @@ object GoogleMapHelper {
...
+        title: String?,
+        draggable: Boolean,
+        idDrawable: Int,
+        leftBound: Int,
+        topBound: Int,
+        rightBound: Int,
+        bottomBound: Int,
+        width: Int,
+        height: Int
+    ): MarkerOptions {
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:397]: Is it possible to group all this parameter in some kind of objects ? Because I think that the parameter list is way too long

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 4]
```diff
@@ -261,46 +282,155 @@ object GoogleMapHelper {
...
-    private fun getMarkerRessource(id: Int): BitmapDescriptor {
+    private fun getMarkerRessource(
+        id: Int,
+        leftBound: Int,
+        topBound: Int,
+        rightBound: Int,
+        bottomBound: Int,
+        width: Int,
+        height: Int
+    ): BitmapDescriptor {
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:430]: Same comment as above, you could maybe try to make an object "BoundAndDimension" or something of this kind so that we only get one parameter instead of 6

File: projectFilesBackup/.idea/workspace.xml [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+    <shared>
+      <config />
+    </shared>
+  </component>
+  <component name="AndroidLogFilters">
+    <option name="TOOL_WINDOW_CONFIGURED_FILTER" value="Show only selected application" />
+  </component>
+  <component name="ChangeListManager">
+    <list default="true" id="b395262b-9905-4ad3-96a5-857dfa73d10c" name="Default Changelist" comment="">
+      <change afterPath="$PROJECT_DIR$/app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapAdapter.kt" afterDir="false" />
```
Comment (AlessioVerardo) [inline projectFilesBackup/.idea/workspace.xml:13]: Is this file really needed on the online repo ? :)

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt [hunk 6]
```diff
@@ -422,8 +601,279 @@ object GoogleMapHelper {
...
+     * @param angle: the angle of the rotation
+     * @param center: the center of the rotation in lat/lng coordinates
+     * @return the rotated point in lat/lng coordinates
+     */
+    fun applyRotation(point: LatLng?, angle: Double, center: LatLng): LatLng {
+        val pointCartesian = equirectangularProjection(point, center)
+        val rotatedCartesian = computeRotation(pointCartesian, angle)
+        return inverseEquirectangularProjection(rotatedCartesian, center)
+    }
 }
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt:879]: Nice and clean methods :) 
Maybe they can be added into the MapsInterface

**Review by AlessioVerardo — APPROVED** [review]
Overall really good code :)

File: app/build.gradle (hunk 1 above)
Comment (cpittet) [reply app/build.gradle]: I merged with the main branch so it's good now

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt (hunk 3 above)
Comment (cpittet) [reply app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt]: It should have been fixed by Mathieu's PR

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt (hunk 4 above)
Comment (cpittet) [reply app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt]: Fixed by Mathieu's PR

File: projectFilesBackup/.idea/workspace.xml (hunk 5 above)
Comment (cpittet) [reply projectFilesBackup/.idea/workspace.xml]: Nope, it has also been removed by Mathieu :)

**Review by Rygone — APPROVED** [review]

**Review by Feez2403 — APPROVED** [review]
Great TOUPIE !
